### ISSUE TITLE
Social Media links below "SMALL-FIRM SENSIBILITY WITH LARGE-FIRM CAPABILITIES"

### ISSUE THREAD

**Issue by d4mation** [realbig/RBM_TH_INOSENCIO#8]
Should be editable by client

**Comment by d4mation** [comment]
They noted liking butzellong.com's implementation, but I can't even find a backup of that in the Wayback Machine. Each entry I checked redirects to what it is now, which states it is for sale.

https://web.archive.org/web/20160801000000*/http://butzellong.com

**Comment by d4mation** [comment]
Decided to try Googling "Butzel Long". They provided the _wrong domain_.

https://www.butzel.com/ is the correct one.

They appear to want feeds for each Platform. This may be best accomplished through a 3rd party plugin.

**Comment by d4mation** [comment]
Did some poking around, and this is the plugin in use specifically by Butzel Long. We could always just use the exact same implementation. Mere $21 license.

https://codecanyon.net/item/wordpress-social-stream/2201708

EDIT: Had not-quite the right product in there. The WordPress version of it is $21. Still pretty cheap.

**Comment by d4mation** [comment]
@BigActual What do you think? If we're going to do a Social Media Feed, we're going to want to go 3rd party with this anyhow. Considering they picked out something they liked, we could just have them use precisely what they liked. Maybe tweak the appearance here or there to match with the theme, but otherwise leave it as-is functionality-wise.

**Comment by BigActual** [comment]
Client has the information to purchase and will send details when complete.

**Comment by d4mation** [comment]
Any updates on this?

**Comment by d4mation** [comment]
The Social Feed appears to have 2 Settings: Wall and Feed.

No matter which you choose, you can decide which Social Media platform to put into it and you can place multiple different platforms into a single Wall/Feed. The attached screenshots show with both Facebook and Twitter active within it.

I think the best way to handle this, however, is to have a single platform per **_Feed_** and have each within separate columns. This is actually how Butzel Long's site did it as well. It just will look cleaner.

I'm going to create a Sidebar/Widget Area in the desired location, have the number of columns be adjustable by the client, and then have some Widgets that allow different Feeds/Walls be added into there.

Example of "Wall":
![image](https://user-images.githubusercontent.com/7770631/30593323-25732972-9d18-11e7-9945-685d136d0ca5.png)

Example of "Feed":
![image](https://user-images.githubusercontent.com/7770631/30593348-396f7f52-9d18-11e7-9f67-2e54dafb420b.png)

**Comment by d4mation** [comment]
@BigActual This has been implemented on Staging.

Streams need to be created here: http://inosencio-com.realbig.staging.wpengine.com/wp-admin/options-general.php?page=social-stream

I have not configured this on Staging due to the fact that the Social Platforms that are most likely to be used, Facebook and Twitter, require API Keys in order to function. I used my own personal ones while testing. The Client should utilize their own API Keys for this as to pass off any kind of rate limiting to them (In the event that they need to pay for larger quotas, the keys somehow get blocked/banned, etc.)

After that is done, on the Home Page there are Widget Areas where they can be added. I created a Widget called "Social Stream" which provides a UI to add them so that no one has to deal with Shortcodes.

![image](https://user-images.githubusercontent.com/7770631/30597557-536bfd6a-9d25-11e7-813a-67ff0d2326e1.png)

The number of these Widget Areas can be adjusted within the Customizer. So if having only 1 or 2 makes more sense, that can be quickly changed without needing any code alterations be done.

Like I mentioned above, given this column approach, it will make the most sense to have _one_ Social Media Platform configured per-"Stream". This is how the Butzel Long site has done it and I think it works much better and is more clear than having them all intermingled into a single Feed.
